project1/ImmersiveView.swift

```diff
@@ -70,71 +73,179 @@ class SequenceViewModel: ObservableObject {
-                DispatchQueue.main.async {
-                    switch result {
-                    case .success(let data):
-                        do {
-                            let decodedData = try JSONDecoder().decode(EndpointData.self, from: data)
-                            // Update the model with fetched data
-                            self?.durations = decodedData.durations
-                            self?.loudness = decodedData.loudness
-                            self?.pitches = decodedData.pitches
-                            let value = decodedData.loudness.enumerated().map{ (index, element) in
-                                if(element > -10){
-                                    return decodedData.durations[index]
-                                }else{
-                                    return nil
-                                }
-                            }.compactMap{$0}
-                           print(value)
-                            
-                            // Optionally, trigger actions based on the fetched data
-                            self?.startSequence(times: value)
-                        } catch {
-                            print("Failed to decode data: \(error.localizedDescription)")
-                        }
-                    case .failure(let error):
-                        print("Failed to fetch data: \(error.localizedDescription)")
+        let networkManager = NetworkManager()
+        networkManager.fetchData(from: urlString) { [weak self] result in
+            DispatchQueue.main.async {
+                switch result {
+                case .success(let data):
+                    do {
+                        let decodedData = try JSONDecoder().decode(EndpointData.self, from: data)
+                        // Update the model with fetched data
+                        self?.durations = decodedData.durations
+                        self?.loudness = decodedData.loudness
+                        self?.pitches = decodedData.pitches
+                        let value = decodedData.loudness.enumerated().map{ (index, element) in
+                            if(element > -10){
+                                return decodedData.durations[index]
+                            }else{
+                                return nil
+                            }
+                        }.compactMap{$0}
+                       print(value)
+                        
+                        // Optionally, trigger actions based on the fetched data
+                        self?.startSequence(times: value)
+                    } catch {
+                        print("Failed to decode data: \(error.localizedDescription)")
                     }
+                case .failure(let error):
+                    print("Failed to fetch data: \(error.localizedDescription)")
                 }
             }
         }
+    }
+    
+   func initSequence(randomSeed: Float) {
```

Comment:
This provides a quick way to experiment with particle system parameters, removing the need to wait for Spotify data. While I think real music data is important, at this point it doesn't provide much value, because it only generates random number and adds latency. I will usually call this in `.onAppear` and comment out `viewModel.fetchDataFromEndpoint`.